src/components/side_panel/table/table_panel.ts

```diff
@@ -0,0 +1,151 @@
+    .o-table-n-of-headers {
+      width: 14px;
+      color: #666666;
+      line-height: 1;
+      text-align: center;
+      border: none;
+      border-bottom: 1px solid #ccc;
+      padding: 0px 2px;
+      &:focus {
+        outline: none;
+      }
+    }
+  }
+`;
+
+export class TablePanel extends Component<Props, SpreadsheetChildEnv> {
+  static template = "o-spreadsheet-TablePanel";
+  static components = { TableStylePicker, SelectionInput, ValidationMessages, Checkbox, Section };
+  static props = { onCloseSidePanel: Function };
+
+  state!: State;
+
+  setup() {
+    const selectedTable = this.env.model.getters.getFirstTableInSelection();
+    if (!selectedTable) {
+      this.props.onCloseSidePanel();
+      return;
+    }
+    this.state = useState({
+      table: selectedTable,
+      tableZoneErrors: [],
+      tableXc: zoneToXc(selectedTable.zone),
+    });
+
+    onWillUpdateProps(() => {
+      const selectedTable = this.env.model.getters.getFirstTableInSelection();
+      if (selectedTable) {
+        if (this.state.table.id !== selectedTable.id) {
+          this.state.tableXc = zoneToXc(selectedTable.zone);
+          this.state.tableZoneErrors = [];
+        }
+        this.state.table = selectedTable;
+      } else {
+        this.props.onCloseSidePanel();
+      }
+    });
+  }
+
+  updateTableConfig(attName: keyof TableConfig, value: boolean | string | number): DispatchResult {
+    return this.updateTable({ [attName]: value });
+  }
+
+  updateTable(config?: Partial<TableConfig>, newZone?: Zone): DispatchResult {
+    const sheetId = this.env.model.getters.getActiveSheetId();
+    return this.env.model.dispatch("UPDATE_TABLE", {
+      sheetId,
+      zone: this.state.table.zone,
+      config,
+      newTableZone: newZone,
+    });
+  }
+
+  onChangeNumberOfHeaders(ev: Event) {
+    const input = ev.target as HTMLInputElement;
+    const numberOfHeaders = parseInt(input.value);
+    const result = this.updateTableConfig("numberOfHeaders", numberOfHeaders);
+    if (!result.isSuccessful) {
+      input.value = this.state.table.config.numberOfHeaders.toString();
+    }
+  }
+
+  onRangeChanged(ranges: string[]) {
+    if (!ranges[0].match(rangeReference)) {
+      this.state.tableZoneErrors = [CommandResult.InvalidRange];
+      return;
+    }
+    const sheetId = this.env.model.getters.getActiveSheetId();
+
+    const newTableZone = this.env.model.getters.getRangeFromSheetXC(sheetId, ranges[0]).zone;
+    this.state.tableXc = zoneToXc(newTableZone); // make sure we don't have an unbounded zone
+
+    this.state.tableZoneErrors = this.env.model.canDispatch("UPDATE_TABLE", {
+      sheetId,
+      zone: this.state.table.zone,
+      newTableZone,
+    }).reasons;
+  }
+
+  onRangeConfirmed() {
+    const newZone = toZone(this.state.tableXc);
+    const result = this.updateTable(undefined, newZone);
+
+    const position = { col: newZone.left, row: newZone.top };
+    this.env.model.selection.selectZone({
```

Comment:
I guess my comment about the zone belongs here instead - why should we change the selection upon confirmation?

Reply:
Yeah. The side panel is editing the table that is currently inside the selection. If we move the table, but don't change the selection, the side panel will close itself since there is no table in the selection :x